modules/json/src/smithy4s/json/JsoniterCodecCompiler.scala

```diff
@@ -78,6 +78,12 @@ trait JsoniterCodecCompiler extends CachedSchemaCompiler[JsonCodec] {
     */
   def withLenientTaggedUnionDecoding: JsoniterCodecCompiler
 
+  /**
+    * Enables lenient decoding of numeric values, where numbers may be carried by JSON strings
+    * as well as JSON numbers.
+    */
+  def withLenientNumericDecoding: JsoniterCodecCompiler
```

Comment:
**question:** shall we just call it lenient decoding, in case we add more cases of leniency? such as, case-insensitivity (which might not be doable if there are members with names like `foo` and `FOO`, but might be possible in many cases)

Reply:
and we'd just mention number handling because it happens to be the only thing for now

Reply:
I'd rather have more granular options than a "one size fits all" option.